client-lib/src/connector/timeplus/TimeplusConnector.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable camelcase */
+import { DPMConfiguration, nameToSlug, Parameter, ParameterType } from "datapm-lib";
+import { Connector } from "../Connector";
+import { TYPE } from "./TimeplusConnectorDescription";
+import fetch from "cross-fetch";
+import { JobContext } from "../../task/Task";
+
+const printedTimeplusLoginMessage = false;
+
+export class TimeplusConnector implements Connector {
+    getType(): string {
+        return TYPE;
+    }
+
+    requiresConnectionConfiguration(): boolean {
+        return true;
+    }
+
+    userSelectableConnectionHistory(): boolean {
+        return true;
+    }
+
+    requiresCredentialsConfiguration(): boolean {
+        return true;
+    }
+
+    async getRepositoryIdentifierFromConfiguration(connectionConfiguration: DPMConfiguration): Promise<string> {
+        if (typeof connectionConfiguration.host !== "string") {
+            throw new Error("Timeplus host not set");
+        }
+        return connectionConfiguration.host;
+    }
+
+    async getCredentialsIdentifierFromConfiguration(
+        _connectionConfiguration: DPMConfiguration,
+        credentialsConfiguration: DPMConfiguration
+    ): Promise<string | undefined> {
+        if (credentialsConfiguration.token != null) {
+            const token = credentialsConfiguration.token as string;
```

Comment:
I'll accept this PR, but it would be ideal if it could return the username of the authenticated user.

Reply:
sorry, at this point the token is shared for the tenant, not per user (working on it)